### PR TITLE
do issue related to basic syntax (#35)

### PR DESCRIPTION
**Pull request by raminLgh**
closes: #35

### CODE REVIEW

File: ramin/js/basic-syntax-35/ramin.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+// concat two strings
+function concat(stringA, stringB) {
+    return stringA + ' ' + stringB;
+}
+
+/// ----------------------------------- ////
+
+// 2^x = ?
+function power2(x) {
+    return Math.pow(x,2);
```
Comment (mkermani144) [inline ramin/js/basic-syntax-35/ramin.js:30]: This is wrong.

File: ramin/js/basic-syntax-35/ramin.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
+// add number a to b
+function add(a, b) {
+    return a+b;
+}
+
+/// ----------------------------------- ////
+
+// flip a coin and log the result (head or tail)
+function flipACoin() {
+    //Math.random() generate a 0<=floating_point_number<1
```
Comment (mkermani144) [inline ramin/js/basic-syntax-35/ramin.js:10]: Do not leave these kinds of comments in your code. Anyone familiar with JS knows what `Math.random` does.

File: ramin/js/basic-syntax-35/ramin.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    return Math.pow(x,2);
+}
+
+/// ----------------------------------- ////
+
+// user object with first name and last name, both capitalized
+// e.g. getUser('hello', 'world') is { firstName: 'Hello', lastName: 'World' }
+function getUser(firstName, lastName) {
+
+    let Uper_firstName = firstName[0].toUpperCase() + firstName.slice(1); // if firstName = Ramin => R + amin
```
Comment (mkermani144) [inline ramin/js/basic-syntax-35/ramin.js:39]: Variable names should be camelCase. In addition, please remove the redundant code.

File: ramin/js/basic-syntax-35/ramin.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+// e.g. greet('ConnecMent') is "Hello, ConnecMent! Hope you the best."
+function greet(name) {
+    console.log(`Hello, ${name}!`);
+}
+
+/// ----------------------------------- ////
+
+// get array of n random numbers
+function getNRandomNumberss(n) {
+    let arr_num = [];
```
Comment (mkermani144) [inline ramin/js/basic-syntax-35/ramin.js:59]: Same here. The convention in JS for variable names is camelCase, not snake_case.

File: ramin/js/basic-syntax-35/ramin.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+}
+
+/// ----------------------------------- ////
+
+// get sorted array of n random numbers, in range min to max
+function getNRandomNumbers(n, min, max) {
+
+    let arr_num = []
+    let seed = max - min;
```
Comment (mkermani144) [inline ramin/js/basic-syntax-35/ramin.js:75]: Why is this called a _seed_?

File: ramin/js/basic-syntax-35/ramin.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    console.log(time);
+
+    let id = setInterval(function print_time(){
+        let date1 = new Date();
+        let time = date1.toLocaleTimeString();
+        console.log(time)
+
+        ++count;
+        
+        if(count === n/x){
```
Comment (mkermani144) [inline ramin/js/basic-syntax-35/ramin.js:104]: There is a minor bug here. If n is not a multiplier of x (e.g. x=2, n=3), this condition is never met and the inteval runs forever.

File: ramin/js/basic-syntax-35/ramin.js [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+            clearInterval(id);
+        }
+
+    },x*1000);
+
+}
+
+
+//logCurrentTime(2,10)
```
Comment (mkermani144) [inline ramin/js/basic-syntax-35/ramin.js:113]: Do something for this line of code. Either remove it, or uncomment it.

File: ramin/js/basic-syntax-35/ramin.js [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+    for(let i=0;i<n;++i){
+        func(...args);
+    }
+}
+
+/// ----------------------------------- ////
+
+// return true if parameter is falsy, false otherwise
+function isNil(param) {
+    return !Boolean(param);
```
Comment (mkermani144) [inline ramin/js/basic-syntax-35/ramin.js:129]: You can omit `Boolean(...)` as the negate operator (`!`) converts the operand to a boolean.

**Review by mkermani144 — CHANGES_REQUESTED** [review]

File: ramin/js/basic-syntax-35/ramin.js (hunk 5 above)
Comment (raminLgh) [reply ramin/js/basic-syntax-35/ramin.js:75]: Hello
Yes, you are right, it is not a good name. I got it wrong with something in c

File: ramin/js/basic-syntax-35/ramin.js (hunk 6 above)
Comment (raminLgh) [reply ramin/js/basic-syntax-35/ramin.js:104]: sorry . I use parseInt() function to fixed this problem.

**Review by mkermani144 — APPROVED** [review]

**Pull request merged**